libsamplerate, libdvdcss: fix darwin build

ryanartecona opened the pull request:
###### Things done:
- [ ] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [x] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

I started with intent to get Handbrake building on darwin, but that turned our more difficult than anticipated. It seems to pretty heavily depend on Xcode.

I did manage to fix these two in the process, though. They build for me locally just fine. The nox-review fails for me while building doxygen, which seems pretty unrelated.

ryanartecona committed:
libdvdcss: add darwin support

joachifm commented on pkgs/development/libraries/libdvdcss/default.nix:
```diff
@@ -1,9 +1,13 @@
-stdenv.mkDerivation rec {
+with stdenv;
+
+mkDerivation rec {
```
Hm, not that it makes a huge difference, but this is a bit unusual. What you typically see is `with stdenv.lib`.

ryanartecona commented:
Anything keeping this from being merged? Any other info I can provide?

FRidh commented:
@ryanartecona considering there weren't any more replies I don't think there is anything that should keep it from merging. Could you rebase and cc me?

ryanartecona commented:
Thanks, @FRidh!

It looks like libsamplerate has been fixed on darwin separately already in e2fdd74a41fb8859f72e4f26636df79b046caee5. I removed the commit which did basically that same thing in this branch.

I rebased the libdvdcss commit onto current master. A local darwin build after the rebase seems to still work.

FRidh commented:
@ryanartecona its just a minor issue, but could you also undo the change that @joachifm pointed at.

ryanartecona commented:
@FRidh Sure thing. Done.
